llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -14935,7 +15108,7 @@ Value *BoUpSLP::vectorizeTree(TreeEntry *E, bool PostponedPHIs) {
       Value *V = Builder.CreateBinOp(
           static_cast<Instruction::BinaryOps>(E->getOpcode()), LHS,
           RHS);
-      propagateIRFlags(V, E->Scalars, VL0, It == MinBWs.end());
+      propagateIRFlags(V, E->Scalars, nullptr, It == MinBWs.end());
```

Comment:
Should it always be nullptr or are there cases where we can keep it?

Reply:
It should pass VL0 here, but need to check if all opcodes are originally mathed and if not, then pass fourth argument `/*IncludeWrapFlags=*/false`

Reply:
Before the PR, there is no difference between VL0 and nullptr. The opcode must be the same for all VL.
After the PR, VL0 cannot be used because opcode may be different.
I don't know why VL0 is used in the beginning since pass VL0 and nullptr will get the same result for propagateIRFlags.

Reply:
I think this part still requires extra work. If the opcode of the instruction does not match the opcode of intersection, its flags are ignored. This is not correct and must be fixed

Reply:
This is used to fix
```
- ; CHECK-NEXT:    [[TMP1:%.*]] = mul nsw <4 x i32> [[SHUFFLE]], <i32 5, i32 9, i32 3, i32 10>
- ; CHECK-NEXT:    [[TMP2:%.*]] = shl <4 x i32> [[SHUFFLE]], <i32 5, i32 9, i32 3, i32 10>
- ; CHECK-NEXT:    [[TMP3:%.*]] = shufflevector <4 x i32> [[TMP1]], <4 x i32> [[TMP2]], <4 x i32> <i32 0, i32 1, i32 6, i32 3>
+ ; CHECK-NEXT:    [[TMP3:%.*]] = mul <4 x i32> [[SHUFFLE]], <i32 5, i32 9, i32 8, i32 10>
```
VL0 is mul here. If we pass VL0, then eventually `nsw` will be passed.
However, shl does not contain `nsw`. We should pass nullptr here to get the correct result.

Reply:
Say, you have something like `{mul nsw, shl, mul nsw, mul nsw}`. For this case you will still emit `mul nsw <4 x >]б b ecause shl will be ignored

Reply:
Yes, it what is expected. But I assume that instead currently it will emit `mul nsw <4 x >`